app/services/teacher_training_public_api/sync_sites.rb

```diff
@@ -69,7 +78,12 @@ def create_course_options(site, study_mode, site_status)
       )
 
       vacancy_status = vacancy_status(site_status.vacancy_status, study_mode)
-      course_option.update!(vacancy_status: vacancy_status) if course_option.vacancy_status != vacancy_status
+
+      if course_option.vacancy_status != vacancy_status
```

Comment:
While testing, this was always raising an error. It turns out the value returned from `vacancy_status` is a symbol, while the `course_option.vacancy_status` is a string, meaning they are never equal, and this is always being updated. I think we can try `course_option.vacancy_status != vacancy_status.to_s` which will make sure we only update when the value has changed.